app/src/main/kotlin/com/simplemobiletools/calculator/operation/CurrencyConversionOperation.kt

```diff
@@ -0,0 +1,33 @@
+package com.simplemobiletools.calculator.operation
+
+import com.beust.klaxon.JsonObject
+import com.beust.klaxon.Parser
+
+class CurrencyConversionOperation(var value: Double, var conversion_from: String,
+                                           var conversion_to: String, conversionRatesJsonString: String) {
+
+    var rates = (Parser().parse(conversionRatesJsonString!!.reader()) as JsonObject).obj("rates")!!
```

Comment:
I feel like the json should be parsed before it comes into here.

Reply:
I agree. Will instead parse the json in the postExecute of the AsyncTask that grabs the API data.